sound/soc/sof/ipc4-topology.c

```diff
@@ -1533,17 +1533,14 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 		out_ref_valid_bits = SOF_IPC4_AUDIO_FORMAT_CFG_V_BIT_DEPTH(in_fmt->fmt_cfg);
 		break;
 	}
-	case snd_soc_dapm_aif_out:
-	case snd_soc_dapm_dai_in:
+	default:
 		out_ref_valid_bits = sof_ipc4_get_valid_bits(sdev, fe_params);
 		if (out_ref_valid_bits < 0)
 			return out_ref_valid_bits;
 
 		out_ref_rate = params_rate(fe_params);
 		out_ref_channels = params_channels(fe_params);
 		break;
-	default:
-		break;
```

Comment:
I find this very very convoluted, with a dependency on what happens in the previous switch. That's dangerous IMHO, if we change the first switch we may break the second without knowing it. The alternative below is much simpler:

````
case snd_soc_dapm_aif_out:
case snd_soc_dapm_dai_in:
	out_ref_valid_bits = sof_ipc4_get_valid_bits(sdev, fe_params);
	if (out_ref_valid_bits < 0)
		return out_ref_valid_bits;
	out_ref_rate = params_rate(fe_params);
	out_ref_channels = params_channels(fe_params);
default:
    return -EINVAL;
````

That way both switch cases have a consistent behavior but remain independent.

Reply:
@plbossart it is makes you feel less paranoid i'm good with the alternate suggestion but it just cannot happen. Would a comment make you feel more comfortable instead?

Reply:
> @plbossart it is makes you feel less paranoid i'm good with the alternate suggestion but it just cannot happen. Would a comment make you feel more comfortable instead?

we can add a commit that this cannot happen but leave the return -EINVAL; to make sure the static analysis tools don't complain. That way we make both humans and bots happy.